src/Umbraco.Core/Trees/MenuItemList.cs

```diff
@@ -38,10 +38,11 @@ public MenuItemList(ActionCollection actionCollection, IEnumerable<MenuItem> ite
         /// <param name="hasSeparator"></param>
         /// <param name="textService">The <see cref="ILocalizedTextService"/> used to localize the action name based on its alias</param>
         /// <param name="opensDialog">Whether or not this action opens a dialog</param>
-        public MenuItem? Add<T>(ILocalizedTextService textService, bool hasSeparator = false, bool opensDialog = false)
+        /// <param name="useLegacyIcon">Whether or not this action should use legacy icon prefixed with "icon-" or full icon name is specified.</param>
+        public MenuItem? Add<T>(ILocalizedTextService textService, bool hasSeparator = false, bool opensDialog = false, bool useLegacyIcon = true)
```

Comment:
I'll highlight this one again @nathanwoulfe and @bjarnef - this is a common mistake: you can not add an optional parameter to a public method, this is always a breaking change as the signature of the method is changed. 

Luckily the fix for this is easy and I fixed it here: https://github.com/umbraco/Umbraco-CMS/commit/885ec99cc48d4a3faa8becf8a87914f4a2390c92

So I took the whole method body and added it to a new method with the optional parameter on it. Then I call the new method from the old method (signature is unchanged), with the default value of `useLegacyIcon: false`.

Reply:
@nul800sebastiaan okay, I know it would be a breaking change if the order of the parameters changed or if the new options parameter was added before the existing ones, but didn't thought I would be a breaking change with the new parameter as optional and as last parameter.

Must have missed this method when I checked the custom code with various methods before and after this change.

Reply:
It's a common mistake! All good, fixed it up 👍